src/components/AudioDetails/VAudioDetails.vue

```diff
@@ -41,18 +52,24 @@
               {{ $t('audio-details.table.provider') }}
             </dt>
             <dd>
-              {{ providerName }}
+              <a
+                :href="audio.foreign_landing_url"
+                target="blank"
+                rel="noopener noreferrer"
+              >
+                {{ providerName }}
+              </a>
             </dd>
           </div>
-          <div v-if="audio.source">
+          <div v-if="audio.source && sourceName !== providerName">
             <dt>
               {{ $t('audio-details.table.source') }}
             </dt>
             <dd>
               {{ sourceName }}
             </dd>
           </div>
-          <div v-if="audio.genres">
+          <div v-if="audio.genres.length > 0">
```

Comment:
Genres can sometimes be null. Try this on `/audio/632d0076-fdbd-4919-8b8f-b498ef246ee8` and it will crash on this line.

```suggestion
          <div v-if="audio.genres?.length > 0">
```

Reply:
ah, thanks! assumed it was always an empty array 😢